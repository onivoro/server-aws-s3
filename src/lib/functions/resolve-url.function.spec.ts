import { TS3Params } from "../services/s3.service";
import { resolveUrl } from "./resolve-url.function";

const region: string = 'us-east-2';

describe('resolveUrl', () => {
    it.each([
        [{ Bucket: 'with-slash', Key: 'item.pdf' }],
        [{ Bucket: 'with.dot', Key: 'item.pdf' }],
    ])('', ({ Bucket, Key }: TS3Params) => {
        expect(resolveUrl(region, { Bucket, Key })).toMatchSnapshot();
    });
});