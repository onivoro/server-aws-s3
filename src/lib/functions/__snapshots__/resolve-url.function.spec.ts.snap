// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`resolveUrl  1`] = `"https://with-slash.s3.us-east-2.amazonaws.com/item.pdf"`;

exports[`resolveUrl  2`] = `"https://s3.us-east-2.amazonaws.com/with.dot/item.pdf"`;
